### PR TITLE
Using new maps and initial location logic

### LINKED ISSUE #161: Prevent users from doing 'select location' until they set a location
On the /location page, show an error message when users try to click 'Select Location' without doing at least one of - 
1) search for a store/locate me
2) moving the pin 

Basically landing on that page and straight away selecting location should be prevented

### PR DESCRIPTION
**Pull request by arjun27**
Fixes #161 #129 #51

Removes current location bits from the map internal and into shared state (context) so that the app can use them. This will set us up to also use current location via query parameters (which is a product requirement for Swiggy integration). Also distinguishes between IP location and geolocation since they have different accuracy levels.

Updates submission flow for
1. On location selection page, IP location is not a valid default selection. Getting an accurate location is important and user must search, move marker or use geolocation at this step. 
2. On submit form page, location is now read only. Users land on this page via updating an existing store or via the location selection page.

Drive by: removes unused redux code.

### CODE REVIEW

**Comment by arjun27** [comment]
@Siddharth11 - sorry about the large PR. would be great if you can skim through and review it.

File: src/LocationSelector/index.js [hunk 1]
```diff
@@ -1,39 +1,43 @@
 import React from "react";
 import { geocodeByLatlng, getAddressComponent, isFunction } from "../utils";
-import MapWithSearch from "../MapWithSearch";
+import LocationSelectorMapWithSearch from "../MapsWithSearch/LocationSelectorMap";
 
 class LocationSelector extends React.Component {
+  geocodeAndInformParent = async latLng => {
+    const results = await geocodeByLatlng(latLng);
```
Comment (Siddharth11) [inline src/LocationSelector/index.js:7]: What happens if this api call fails ? Will it break the page ?

Comment (Siddharth11) [reply src/LocationSelector/index.js:7]: even if it doesn't i would suggest to add a try catch around it, and console log the error

File: src/LocationSelector/index.js [hunk 2]
```diff
@@ -1,39 +1,43 @@
...
+        name: "",
+        address: result.formatted_address,
+        place_id: result.place_id,
+        types: result.types,
+        locality: getAddressComponent(
+          result.address_components,
+          "neighborhood"
+        ),
+        city: getAddressComponent(result.address_components, "locality"),
+        country: getAddressComponent(result.address_components, "country")
```
Comment (Siddharth11) [inline src/LocationSelector/index.js:24]: not super necessary, but we can store all these strings `neighborhood`, `locality` , `country` in a an enum. there's a constant.js file in `src`

File: src/Maps/LocationSelectorMap.js [hunk 3]
```diff
@@ -3,17 +3,18 @@ import { GoogleMap, Marker } from "@react-google-maps/api";
 import { dotIcon, markerIcon, icons, isSameLocation } from "../utils";
 import { mapOptions } from "./theme";
 
-const defaultCenter = { lat: 49.281376, lng: -123.111382 };
+const defaultCenter = { lat: 40.281376, lng: -123.111382 };
```
Comment (Siddharth11) [inline src/Maps/LocationSelectorMap.js:6]: Could you add a comment what this lat and lng corresponds to?

File: src/MapsWithSearch/HomepageMap.js [hunk 4]
```diff
@@ -0,0 +1,28 @@
+import React from 'react';
+import LocationSearchControl from './Input';
+import HomepageMap from '../Maps/HomepageMap';
```
Comment (Siddharth11) [inline src/MapsWithSearch/HomepageMap.js:3]: Is there a reason why the current file and the one being imported have same names? 
Do you think it is confusing ?

**Comment by Siddharth11** [comment]
@arjun27 could you give a brief description or link me to a ticket what this PR aims to solve ?

**Comment by arjun27** [comment]
> @arjun27 could you give a brief description or link me to a ticket what this PR aims to solve ?

Yes, sorry about that. Just updated the PR description

File: src/LocationSelector/index.js (hunk 1 above)
Comment (arjun27) [reply src/LocationSelector/index.js:7]: done

File: src/MapsWithSearch/HomepageMap.js (hunk 4 above)
Comment (arjun27) [reply src/MapsWithSearch/HomepageMap.js:3]: hmm I could go with `HomepageMapWithSearch` for the filename, but I think that's too long and the parent directory covers that description. I'll let this be for now, but open to changing if it's confusing

**Pull request merged**